flow/src/test/java/com/sequenceiq/flow/repository/FlowLogRepositoryTest.java

```diff
@@ -62,6 +67,35 @@ void testPurgeFinalizedFlowLogs() {
         assertThat(result).isEqualTo(2);
     }
 
+    @Test
+    void testFindAllByFlowIdsCreatedDesc() {
+        String flowId1 = randomUUID().toString();
+        FlowLog flowLog1 = createFlowLog(flowId1);
+        String flowId2 = randomUUID().toString();
+        FlowLog flowLog2 = createFlowLog(flowId2);
+        String flowId3 = randomUUID().toString();
+        FlowLog flowLog3 = createFlowLog(flowId3);
+        String flowId4 = randomUUID().toString();
+        FlowLog flowLog4 = createFlowLog(flowId4);
+
+        saveFlowLogs(flowLog1, flowLog2, flowLog3, flowLog4);
+
+        Page<FlowLog> result = underTest.findAllByFlowIdsCreatedDesc(Set.of(flowId1, flowId2, flowId3, flowId4),
+                PageRequest.of(0, 4));
+
+        assertThat(result.getContent().size()).isEqualTo(4);
+        Assertions.assertTrue(result.getContent().contains(flowLog1));
+        Assertions.assertTrue(result.getContent().contains(flowLog2));
+        Assertions.assertTrue(result.getContent().contains(flowLog3));
+        Assertions.assertTrue(result.getContent().contains(flowLog4));
+
+        Page<FlowLog> resultPaginated = underTest.findAllByFlowIdsCreatedDesc(Set.of(flowId1, flowId2, flowId3, flowId4),
+                PageRequest.of(0, 2));
+
+        assertThat(resultPaginated.getContent().size()).isEqualTo(2);
+        assertThat(resultPaginated.getTotalElements()).isEqualTo(4);
```

Comment:
This should be separate test. We should verify following :

1. In addition to count, check the flowId3 and flowId4 are being returned. also, hasNext returns true since there are mote records.
2. Based on paging information in result, prepare next page information and again call the API.
3. check that now flowId1 and flowId2 are being returned.also, hasNext returns false since there are no more records.

Reply:
done.